code/cycler/src/wattrex_battery_cycler/app/app_man/app_man_node.py

```diff
@@ -116,30 +119,32 @@ def configure_cs(self, reqs_chan: SysShdChanC, data_chan: SysShdChanC,
         reqs_chan.send_data(request)
         response: MidStrCmdDataC = data_chan.receive_data()
         if response.error_flag:
-            raise ValueError(("Error in response from MID STR"))
-        cs_info = response.station
-        # launch the man_core and meas node if cs is not deprecated
-        if not cs_info.deprecated:
-            ### 1.2 Manager thread ###
-            self.man_core: AppManCoreC= AppManCoreC(devices=cs_info.devices, # pylint: disable=attribute-defined-outside-init
-                                    str_reqs= reqs_chan, str_data= data_chan,
-                                    str_alarms= alarms_chan)
-            ### 1.3 Meas thread ###
-            self._th_meas = MidMeasNodeC(working_flag= self.working_meas, # pylint: disable=attribute-defined-outside-init
-                    shared_gen_meas= self.__shd_gen_meas, shared_ext_meas= self.__shd_ext_meas,
-                    shared_status= self.__shd_all_status, devices= cs_info.devices,
-                    cycle_period= _PERIOD_CYCLE_MEAS, excl_tags= self.__shared_tags)
-            self._th_meas.start()
+            log.error(("Was imposible to get the cycler station info from the database. "))
+            self.status = SysShdNodeStatusE.INTERNAL_ERROR
         else:
-            log.critical(("Cycler station is deprecated. Meas node will not be launched, "
-                        "cycler station will be stop"))
-            self.stop()
-        sleep(2)
-        self.iter = -1 # pylint: disable=attribute-defined-outside-init
-        self.sync_shd_data(raised_alarms= [])
-        while self._th_meas.status != SysShdNodeStatusE.OK:
-            sleep(1)
-        self.status = SysShdNodeStatusE.OK
+            cs_info = response.station
+            # launch the man_core and meas node if cs is not deprecated
+            if not cs_info.deprecated:
+                ### 1.2 Manager thread ###
+                self.man_core: AppManCoreC= AppManCoreC(devices=cs_info.devices, # pylint: disable=attribute-defined-outside-init
+                                        str_reqs= reqs_chan, str_data= data_chan,
+                                        str_alarms= alarms_chan)
+                ### 1.3 Meas thread ###
+                self._th_meas = MidMeasNodeC(working_flag= self.working_meas, # pylint: disable=attribute-defined-outside-init
+                        shared_gen_meas= self.__shd_gen_meas, shared_ext_meas= self.__shd_ext_meas,
+                        shared_status= self.__shd_all_status, devices= cs_info.devices,
+                        cycle_period= _PERIOD_CYCLE_MEAS, excl_tags= self.__shared_tags)
+                self._th_meas.start()
+            else:
+                log.critical(("Cycler station is deprecated. Meas node will not be launched, "
```

Comment:
YOu must perform the same stop as in the previous case when CS was not found, line 104.

Reply:
I have improve this function, if there is a problem that causes cs to stop, after the configure cs function if the status is not ok will stop str node.